### PR TITLE
[ONNX] Fix 1d case flatten export

### LINKED ISSUE #74142: [Bug][ONNX] Specification Inconsistency in Flatten
### 🐛 Describe the bug

```python
import torch

N = 1 # can be anything else

class Net(torch.nn.Module):
    def __init__(self) -> None:
        super().__init__()
    
    def forward(self, x):
        return x.flatten()

net = Net().eval()

o = net(torch.zeros((N), dtype=torch.float))
print(o.shape)

with torch.no_grad():
  torch.onnx.export(net, (
      torch.ones((N), dtype=torch.float)), "output.onnx", verbose=True, opset_version=14)
```

## In ONNX (Also ONNXRuntime)

Flatten returns a shape of (1, N).

image

## However in Torch

Flatten returns a shape of (N).

image

### Versions

PyTorch version: 1.10.2+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04) 9.4.0
Clang version: 10.0.0-4ubuntu1 
CMake version: version 3.22.2
Libc version: glibc-2.31

Python version: 3.8.12 (default, Oct 12 2021, 13:49:34)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.13.0-30-generic-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1060 3GB
Nvidia driver version: 510.47.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.3
[pip3] torch==1.10.2
[conda] numpy                     1.22.2                   pypi_0    pypi
[conda] torch                     1.10.2                   pypi_0    pypi

cc: @garymm @BowenBao @neginraoof @thiagocrepaldi

### PR DESCRIPTION
**Pull request by BowenBao**
Fixes #74142

Previous check `dim is not None and end_dim == dim - 2` didn't consider `end_dim` being negative. However the error only occurs when input tensor has rank 1, and the rank is known to symbolic function. So a better fix is to return `input` directly when rank is 1.

### CODE REVIEW

**Review by garymm — APPROVED** [review]
@pytorchbot merge this.
